frontend/components/password-change/js/password-change.component.ts

```diff
@@ -80,13 +80,19 @@ class PasswordChangeController implements ng.IController {
 
 	public watchers() {
 
-		this.$scope.$watch("vm.newPassword", () => {
+		this.$scope.$watchGroup(["vm.newPassword", "vm.confirmPassword"], () => {
 			this.message = "";
 			if (this.newPassword !== undefined) {
 				const result = this.PasswordService.evaluatePassword(this.newPassword);
 				this.passwordStrength = this.PasswordService.getPasswordStrength(this.newPassword, result.score);
 				this.checkInvalidPassword(result);
 			}
+			if (this.confirmPassword !== undefined && this.confirmPassword === this.newPassword) {
+				this.confirmPasswordInput();
+				this.buttonDisabled = false;
```

Comment:
I think **buttonDisabled** state should be based on both confirmPasswordInput and checkInvalidPassword().

At the moment, if you enter a weak password (e.g. "asdf") and then do the confirm password correctly, you can press the "Set Password" button, which should be disabled because the password is still weak.